hibernate-core/src/test/java/org/hibernate/orm/test/annotations/collectionelement/OrderByTest.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.orm.test.annotations.collectionelement;
+
+import java.util.HashSet;
+import java.util.Iterator;
+
+import org.hibernate.Transaction;
+import org.hibernate.dialect.TeradataDialect;
+
+import org.hibernate.testing.orm.junit.DomainModel;
+import org.hibernate.testing.orm.junit.SessionFactory;
+import org.hibernate.testing.orm.junit.SessionFactoryScope;
+import org.hibernate.testing.orm.junit.SkipForDialect;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+@DomainModel(
+		annotatedClasses = {
+				Products.class,
+				Widgets.class,
+				BugSystem.class
+		}
+)
+@SessionFactory
+public class OrderByTest {
+
+	@Test
+	public void testOrderByName(SessionFactoryScope scope) {
+		scope.inSession(
+				session -> {
+					Transaction tx = session.beginTransaction();
+					try {
+
+						Products p = new Products();
+						HashSet<Widgets> set = new HashSet<>();
+
+						Widgets widget = new Widgets();
+						widget.setName( "hammer" );
+						set.add( widget );
+						session.persist( widget );
+
+						widget = new Widgets();
+						widget.setName( "axel" );
+						set.add( widget );
+						session.persist( widget );
+
+						widget = new Widgets();
+						widget.setName( "screwdriver" );
+						set.add( widget );
+						session.persist( widget );
+
+						p.setWidgets( set );
+						session.persist( p );
+						tx.commit();
+
+						tx = session.beginTransaction();
+						session.clear();
+						p = session.get( Products.class, p.getId() );
+						assertTrue( p.getWidgets().size() == 3, "has three Widgets" );
+						Iterator iter = p.getWidgets().iterator();
+						assertEquals( "axel", ( (Widgets) iter.next() ).getName() );
+						assertEquals( "hammer", ( (Widgets) iter.next() ).getName() );
+						assertEquals( "screwdriver", ( (Widgets) iter.next() ).getName() );
+						tx.commit();
+					}
+					catch (Exception e) {
+						if ( tx.isActive() ) {
+							tx.rollback();
```

Comment:
Just curious. Does it make any difference if we rollback or not in testing case? Anyway, each new testing case should be self-contained and independent and not subjected to the running result of last testing case.

Reply:
in case an exception occurred I do not want to leave a transaction active.

Reply:
I think if we use the following pattern the transaction rollback will be done automatically for us:
```
scope.inTransaction(
        session -> {
             // code snippet which might throw exception 
        }
);
```

Reply:
Thanks @NathanQingyangXu fixed it